examples/tv-casting-app/APIs.md

```diff
@@ -0,0 +1,692 @@
+[example Matter tv-casting-app](https://github.com/project-chip/connectedhomeip/tree/master/examples/tv-casting-app)
+samples so you can see the experience end to end.
+
+## Introduction
+
+A Casting Client (e.g. a mobile phone app) is expected to be a Matter
+Commissionable Node and a `CastingPlayer` (i.e. a TV) is expected to be a Matter
+Commissioner. In the context of the
+[Matter Video Player architecture](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc),
+a `CastingPlayer` would map to
+[Casting "Video" Player](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc#1-introduction).
+The `CastingPlayer` is expected to be hosting one or more `Endpoints` (similar
+to
+[Content Apps](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc#1-introduction)
+in the Matter Video Player architecture) that support one or more Matter Media
+`Clusters`.
+
+The steps to start a casting session are:
+
+1. [Initialize](#initialize-the-casting-client) the `CastingClient` using the
+   Matter SDK.
+1. [Discover](#discover-casting-players) `CastingPlayer` devices using Matter
+   Commissioner discovery.
+1. [Connect](#connect-to-a-casting-player) to the `CastingPlayer` to discover
+   available endpoints. By connecting, the 'CastingClient' is first sending a
+   User Directed Commissioning (UDC) request to the 'CastingPlayer' device in
+   order to request Matter commissioning by it. For commissioning, the
+   'CastingPlayer' will obtain the appropriate user consent to allow a
+   connection from this 'CastingClient' and obtain the setup code needed to
+   commission the 'CastingClient'. The setup code will typically come from a
+   corresponding TV content app or be input by the user.
+1. [Select](#select-an-endpoint-on-the-casting-player) an available `Endpoint`
+   hosted by the `CastingPlayer`.
+
+Next, you're ready to:
+
+1. [Issue commands](#issuing-commands) to the `Endpoint`.
+1. [Read](#read-operations) endpoint attributes like playback state.
+1. [Subscribe](#subscriptions) to playback events.
+
+## Build and Setup
+
+The Casting Client is expected to consume the Matter TV Casting library built
+for its respective platform to consume any of the APIs described in this
+document. Refer to the tv-casting-app READMEs for [Android](android) and
+[iOS](darwin) to understand how to build and consume each platform's specicific
+libraries.
+
+### Initialize the Casting Client
+
+_{Complete Initialization examples:
+[Android](android/App/app/src/main/java/com/matter/casting/InitializationExample.java)
+| [iOS](darwin/TvCasting/TvCasting/TvCastingApp.swift)}_
+
+A Casting Client must first initialize the Matter SDK and define the following
+[`DataProvider`](android/App/app/src/main/jni/com/matter/casting/params/DataProvider.java)
```

Comment:
this one didn't work either, maybe just preview is broken? Other links work fine though

https://github.com/sharadb-amazon/connectedhomeip/blob/9e60120c433a864fa7af897ed919d47cf4276c64/examples/tv-casting-app/APIs.md